internal/docstore/mongodocstore/mongo.go

```diff
@@ -687,13 +699,21 @@ func (c *collection) unorderedBulkWrite(ctx context.Context, iactions []indexedA
 		}
 		return alerr
 	}
+	if res.InsertedCount != nCreate {
```

Comment:
See above. I believe that if InsertedCount != nCreate, we would have gotten an error back from the RPC. Do you have a test case that shows otherwise?

Reply:
Added a test.